scripts/metalk8s_cli/mixins/salt.py

```diff
@@ -0,0 +1,119 @@
+import json
+import shlex
+import subprocess
+
+import six
+
+from metalk8s_cli.exceptions import CommandError, CommandInitError
+
+
+class SaltCommandMixin(object):
+    def __init__(self, args):
+        super(SaltCommandMixin, self).__init__(args)
+
+    def run_salt_minion(
+        self, command, local=False, outputter=None, color=True,
+        saltenv=None, pillar=None,
+    ):
+        full_cmd = ['salt-call']
+
+        if local:
+            full_cmd.append('--local')
+        if outputter:
+            full_cmd.extend(['--out', outputter])
+        if color:
+            full_cmd.append('--force-color')
+
+        full_cmd.extend(self._build_args(command))
+
+        if saltenv is not None:
+            full_cmd.append('saltenv={}'.format(saltenv))
+        if pillar is not None:
+            full_cmd.append("pillar='{}'".format(json.dumps(pillar)))
+
+        try:
+            return subprocess.check_output(full_cmd, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as exc:
+            raise CommandError(
+                'Failed to execute "{}" [retcode {}]:\n{}'.format(
+                    ' '.join(full_cmd), exc.returncode, exc.output
+                )
+            )
+
+    def run_salt_master(self, command):
+        # TODO: add cache for the salt-master container ID
```

Comment:
I don't know if we can/want really to have cache for this.
Because if we cache this we need either
1- a variable to "force caching again" 
2- something to check that the container ID still good (which is basically the command executed in `get_salt_master_container`)

Reply:
Indeed, I was actually thinking about giving control to the caller (same as for `saltenv` in `run_salt_minion`, callers store it once and can pass it to each call). This way, if a command expects the container ID to have changed, it can refresh its value, while otherwise saving the cost of running the subprocess again.